applications/nrf_desktop/configuration/nrf52840dk_nrf52840/settings_loader_def.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/* This configuration file defines modules that need to be loaded before
+ * calling settings_load.
+ */
+
+/* This structure enforces the header file is included only once in the build.
+ * Violating this requirement triggers a multiple definition error at link time.
+ */
+
+const struct {} settings_loader_def_include_once;
+
+#include <caf/events/module_state_event.h>
+
+MODULE_ID_PTR_VAR_EXTERN_DEC(main);
```

Comment:
Maybe we could rely on module indexes here? The settings loader uses a bitmask anyway and we could use `module_id_get(size_t idx)` to convert from idx to id.

Reply:
Because of macro limitations discussed offline, we could allow user to provide function that fills the required module flags (`struct module_flags`). In the function user could simply use `MODULE_IDX(main)` (because of the macro usage limitations, we cannot use the `MODULE_ID`/`MODULE_IDX` macro directly to initialize static array).

Thanks to that we could not longer access internal macros of module_state_event and we would no longer need to "duplicate" macro calls related to a single module.

Reply:
If we introduce a way to convert module ID to module idx, we could also avoid traversing through whole array of required modules here.

Reply:
I've applied changes according to what we discussed offline. User will now be requested to provide static inline function implementation that would update bitmask of requested modules indexes. I still keep this PR as draft as it lacks of documentation update.